source/MaterialXView/Editor.cpp

```diff
@@ -244,19 +234,8 @@ void PropertyEditor::addItemToForm(const mx::UIPropertyItem& item, const std::st
             comboBox->setCallback([path, viewer, enumValues](int v)
             {
                 MaterialPtr material = viewer->getSelectedMaterial();
-                mx::ShaderPort* uniform = material ? material->findUniform(path) : nullptr;
-                if (uniform)
-                {
-                    material->getShader()->bind();
-                    if (v < (int) enumValues.size())
-                    {
-                        material->getShader()->setUniform(uniform->getVariable(), enumValues[v]->asA<int>());
-                    }
-                    else
-                    {
-                        material->getShader()->setUniform(uniform->getVariable(), v);
-                    }
-                }
+                const int intv = v < (int)enumValues.size() ? enumValues[v]->asA<int>() : v;
+                material->setUniformInt(path, intv);
```

Comment:
This will not work correctly for enumerations as you are sending over the integer value which matches a string enumeration, but not the string itself. e.g. if you have this paramter on image:
```
    <parameter name="uaddressmode" type="string" value="periodic" enum="constant,clamp,periodic,mirror" uiname="Address Mode U"/>
```
Then if the choise is "constant" this is sending 0 to the shader but also the parameter is being set to "0" instead of "constant".
I assume it's not hard to send over an additional string and create a setUniformEnum(path, int, string) ?

Reply:
seems that the code only expects int types, but I can refactor to a more generic way. 
```
// Integer input. Can map to a combo box if an enumeration
    if (value->isA<int>())
    {
```